Data/ByteString.hs

```diff
@@ -889,6 +891,14 @@ take n ps@(BS x l)
     | otherwise = BS x n
 {-# INLINE take #-}
 
+-- | 'takeEnd' @n xs@ is equivalent to @'drop' ('length' xs - n) xs@.
+takeEnd :: Int -> ByteString -> ByteString
+takeEnd n ps@(BS _ len)
+    | n <= 0    = empty
+    | n >= len  = ps
+    | otherwise = drop (len - n) ps
```

Comment:
GHC is not smart enough to understand that at this point `len - n` is guaranteed to be between 0 and `len`, so invoking `drop` will execute two redundant comparisons. Could you please inline the definition of `drop` here?

Reply:
Maybe a good idea to also inline `take`'s `otherwise` case in `dropEnd`, then?